core/src/Factory.sol

```diff
@@ -111,19 +118,48 @@ contract Factory {
     }
 
     /*//////////////////////////////////////////////////////////////
-                               INCENTIVES
+                               REFERRALS
     //////////////////////////////////////////////////////////////*/
 
-    function enrollCourier(uint32 id, address wallet, uint16 cut) external {
+    /**
+     * @notice Enrolls `msg.sender` in the referral program. This allows frontends/wallets/apps to
+     * credit themselves for a given user's deposit, and receive a portion of their interest. Note
+     * that after enrolling, `msg.sender` will not be eligible for `REWARDS_TOKEN` rewards.
+     * @dev See `Lender.creditCourier`
+     * @param id A unique identifier for the courier
+     * @param cut The portion of interest the courier will receive. Should be in the range [0, 10000),
+     * with 10000 being 100%.
+     */
+    function enrollCourier(uint32 id, uint16 cut) external {
         // Requirements:
         // - `id != 0` because 0 is reserved as the no-courier case
         // - `cut != 0 && cut < 10_000` just means between 0 and 100%
         require(id != 0 && cut != 0 && cut < 10_000);
         // Once an `id` has been enrolled, its info can't be changed
         require(couriers[id].cut == 0);
 
-        couriers[id] = Courier(wallet, cut);
+        couriers[id] = Courier(msg.sender, cut);
+        isCourier[msg.sender] = true;
+
+        emit EnrollCourier(id, msg.sender, cut);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                REWARDS
+    //////////////////////////////////////////////////////////////*/
+
+    function claimRewards(Lender[] calldata lenders, address beneficiary) external returns (uint256 earned) {
+        // Couriers cannot claim rewards because the accounting isn't quite correct for them -- we save gas
+        // by omitting a `Rewards.updateUserState` call for the courier in `Lender._burn`
+        require(!isCourier[msg.sender]);
+
+        unchecked {
+            uint256 count = lenders.length;
+            for (uint256 i = 0; i < count; i++) {
+                earned += lenders[i].claimRewards(msg.sender);
+            }
+        }
 
-        emit EnrollCourier(id, wallet, cut);
+        REWARDS_TOKEN.safeTransfer(beneficiary, earned);
     }
```

Comment:
It wasn't strictly necessary that I move this here, but it's convenient because it means the `isCourier` mapping can be accessed locally, and also lets users claim rewards from multiple `Lender`s at once.